.github/workflows/docker.yml

```diff
@@ -0,0 +1,79 @@
+name: Create and publish a Docker images
+
+on:
+  push:
+    tags:
+      - "v*.*.*"
+    branches:
+      - "master"
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    name: PHP ${{ matrix.php-version }}, Fixer ${{ github.ref == 'refs/heads/master' && 'edge' || github.ref_name }}
+    if: github.repository == 'PHP-CS-Fixer/PHP-CS-Fixer'
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - php-version: '7.4'
+            alpine-version: '3.16'
+          - php-version: '8.0'
+            alpine-version: '3.16'
+          - php-version: '8.1'
+            alpine-version: '3.18'
+          - php-version: '8.2'
+            alpine-version: '3.18'
+          - php-version: '8.3'
+            alpine-version: '3.18'
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```

Comment:
what magic this 65b78e6e13532edd9afa3aa52ac7964289d1a9c1 is?

Reply:
I believe it's a commit's SHA which effectively locks the action in a certain point.

Reply:
can we use v3? https://github.com/docker/login-action/tree/v3

Reply:
I am not sure if it freezes the action at a certain point or if it can change in time. But anyway it should be backward-compatible, right 🤔?

Reply:
v3 is BC, but also accepting patches and minors.
concrete git-sha is pinning to concrete version (like v3.0.0) and won't approve any bugfixes etc.

I'm not sure of value to pinning for specific commit (eg we recommend installing Fixer ^3.40 and not `==3.40.0`), while I find pinning to v3 usage way more readable than specific git-sha

Reply:
You don't need to explain it to me, I know how it works 😅. I am just wondering it _this particular action_ is semver-compatible and pinning to v3 is stable enough.

Reply:
my understanding is "it is", looking at their repo.
but not expert of that repo.

Reply:
Makes sense to me, I just copied it from some documentation. Now all actions are pined to a major version